nifi-nar-bundles/nifi-camel-bundle/nifi-camel-processors/src/main/java/org/apache/nifi/processors/camel/CamelProcessor.java

```diff
@@ -0,0 +1,240 @@
+                session.putAttribute(flowFile, "camelRouteException", exchange.getException().getMessage());
+            }
+            session.transfer(flowFile, FAILURE);
+        }
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        this.descriptors = ImmutableList.of(CAMEL_SPRING_CONTEXT_FILE_PATH, CAMEL_SPRING_CONTEXT_DEF,
+                                            CAMEL_ENTRY_POINT_URI,EXT_LIBRARIES);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        if (getCamelContext() == null
+            || getCamelContext().getStatus()==ServiceStatus.Stopped
+            || getCamelContext().getStatus()==ServiceStatus.Stopping) {
+            try {
+                String camelContextDef=context
+                    .getProperty(CAMEL_SPRING_CONTEXT_DEF).getValue();
+                String camelContextPath=context
+                    .getProperty(CAMEL_SPRING_CONTEXT_FILE_PATH).getValue();
+                ApplicationContext applicationContext;
+                if(camelContextDef!=null && camelContextDef.trim().length()>0){
+                    applicationContext= new GenericXmlApplicationContext(new ByteArrayResource(camelContextDef.getBytes(),"InMemory Camel COntext Definition"));
+                }else{
+                    if(camelContextPath.startsWith("classpath:")){
+                        applicationContext=new ClassPathXmlApplicationContext(camelContextPath.split(":")[1]);
+                    }else{
+                        applicationContext=new FileSystemXmlApplicationContext(camelContextPath.split(":")[1]);
+                    }
+                }
+                camelContext = new SpringCamelContext(applicationContext);
+                camelContext.setApplicationContextClassLoader(new GroovyClassLoader(camelContext.getApplicationContextClassLoader()));
+                camelContext.addStartupListener(new CamelContextStartupListener(getLogger()));
+                ShutdownStrategy shutdownStrategy=new DefaultShutdownStrategy();
+                shutdownStrategy.setTimeout(1);
+                shutdownStrategy.setTimeUnit(TimeUnit.SECONDS);
+                camelContext.setShutdownStrategy(shutdownStrategy);
+                camelContext.start();
+                getLogger().info("Camel Spring Context initialized");
+            } catch (Exception exception) {
+                getLogger().error("Failed to Shutdown Camel Spring Context", exception);
+            }
+        }
+
+    }
+
+    @OnStopped
+    public void stopped(){
+        if (getCamelContext() != null) {
+            try {
+                getCamelContext().stop();
+                getCamelContext().destroy();
+                ((AbstractApplicationContext)getCamelContext().getApplicationContext()).close();
```

Comment:
Agreed. But  I think, that shutdown will be quite abrupt. So chosen to invoke a stop & close. Please correct me otherwise.